azure/services/bastionhosts/spec.go

```diff
@@ -40,69 +41,65 @@ type AzureBastionSpec struct {
 	EnableTunneling bool
 }
 
-// AzureBastionSpecInput defines the required inputs to construct an azure bastion spec.
-type AzureBastionSpecInput struct {
-	SubnetName   string
-	PublicIPName string
-	VNetName     string
-}
-
-// ResourceName returns the name of the bastion host.
-func (s *AzureBastionSpec) ResourceName() string {
-	return s.Name
-}
-
-// ResourceGroupName returns the name of the resource group.
-func (s *AzureBastionSpec) ResourceGroupName() string {
-	return s.ResourceGroup
-}
-
-// OwnerResourceName is a no-op for bastion hosts.
-func (s *AzureBastionSpec) OwnerResourceName() string {
-	return ""
+// ResourceRef implements azure.ASOResourceSpecGetter.
+func (s *AzureBastionSpec) ResourceRef() *asonetworkv1.BastionHost {
+	return &asonetworkv1.BastionHost{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.Name,
+			Namespace: s.Namespace,
+		},
+	}
 }
 
-// Parameters returns the parameters for the bastion host.
-func (s *AzureBastionSpec) Parameters(ctx context.Context, existing interface{}) (parameters interface{}, err error) {
-	if existing != nil {
-		if _, ok := existing.(armnetwork.BastionHost); !ok {
-			return nil, errors.Errorf("%T is not an armnetwork.BastionHost", existing)
-		}
-		// bastion host already exists
-		return nil, nil
+// Parameters implements azure.ASOResourceSpecGetter.
+func (s *AzureBastionSpec) Parameters(ctx context.Context, existingBastionHost *asonetworkv1.BastionHost) (parameters *asonetworkv1.BastionHost, err error) {
+	bastionHost := &asonetworkv1.BastionHost{}
+	bastionHost.Spec = asonetworkv1.BastionHost_Spec{}
+	if existingBastionHost != nil {
+		bastionHost.Spec = existingBastionHost.Spec
```

Comment:
I think it is best to copy over the complete configuration including `ObjectMeta` (which includes Labels, Annotations etc), `TypeMeta`, `Spec`, and `Status`. 
And then later down the flow, CAPZ enforces fields it cares about. 

```suggestion
		bastionHost = existingBastionHost
```

I don't think the `Diff` func will raise a diff on copying over `ObjectMeta`, `TypeMeta`, and `Status`. Have you seen otherwise in your testing?